AK/StringBase.cpp

```diff
@@ -79,6 +74,16 @@ bool StringBase::operator==(StringBase const& other) const
     return bytes() == other.bytes();
 }
 
+ErrorOr<u8*> StringBase::replace_with_uninitialized_buffer(size_t byte_count)
```

Comment:
I'm not sure about this operation. after we call it, none of the invariants of the class hold until data is actually written to by the base class factory function. 

Can we lower more of it so that we don't have a state that's possible where the StringBase has a bunch of capacity and size, but is filled with garbage?

Reply:
Made them private and instead exposed `replace_with_new_{short_,}string`, which take callback that should fill a buffer.